autometa/taxonomy/ncbi.py

```diff
@@ -60,53 +62,79 @@ def __init__(self, dirpath, verbose=False):
-        if os.path.exists(nr_dmnd_fpath):
-            self.nr_fpath = nr_dmnd_fpath
-        if "dmnd" not in os.path.basename(self.nr_fpath):
-            # This check should probably be in the dependencies/databases file...
-            logger.warning(
-                f"DatabaseWarning: {self.nr_fpath} needs to be formatted for diamond!"
-            )
-        # Setup data structures from nodes.dmp, names.dmp and merged.dmp
         self.nodes = self.parse_nodes()
         self.names = self.parse_names()
-        self.merged = self.parse_merged()
-        self.delnodes = self.parse_delnodes()
+        if self.dbType == "ncbi":
+            self.merged_fpath = os.path.join(self.dirpath, "merged.dmp")
+            self.delnodes_fpath = os.path.join(self.dirpath, "delnodes.dmp")
+            self.merged = self.parse_merged()
+            self.delnodes = self.parse_delnodes()
+            # Set prot.accession2taxid filepath
+            self.accession2taxid_fpath = os.path.join(
+                self.dirpath, "prot.accession2taxid"
+            )
+            acc2taxid_gz = ".".join([self.accession2taxid_fpath, "gz"])
+            if not os.path.exists(self.accession2taxid_fpath) and os.path.exists(
+                acc2taxid_gz
+            ):
+                self.accession2taxid_fpath = acc2taxid_gz
+            # Set prot.accession2taxid.FULL.gz filepath
+            self.accession2taxidfull_fpath = os.path.join(
+                self.dirpath, "prot.accession2taxid.FULL"
+            )
+            acc2taxid_gz = ".".join([self.accession2taxidfull_fpath, "gz"])
+            if not os.path.exists(self.accession2taxidfull_fpath) and os.path.exists(
+                acc2taxid_gz
+            ):
+                self.accession2taxidfull_fpath = acc2taxid_gz
+            # Set dead_prot.accession2taxid filepath
+            self.dead_accession2taxid_fpath = os.path.join(
+                self.dirpath, "dead_prot.accession2taxid"
+            )
+            acc2taxid_gz = ".".join([self.dead_accession2taxid_fpath, "gz"])
+            if not os.path.exists(self.dead_accession2taxid_fpath) and os.path.exists(
+                acc2taxid_gz
+            ):
+                self.dead_accession2taxid_fpath = acc2taxid_gz
+            # Check formatting for nr database
+            self.nr_fpath = os.path.join(self.dirpath, "nr.gz")
+            nr_bname = os.path.splitext(os.path.basename(self.nr_fpath))[0]
+            nr_dmnd_fname = ".".join([nr_bname, "dmnd"])
+            nr_dmnd_fpath = os.path.join(self.dirpath, nr_dmnd_fname)
+            if os.path.exists(nr_dmnd_fpath):
+                self.nr_fpath = nr_dmnd_fpath
+            if "dmnd" not in os.path.basename(self.nr_fpath):
+                # This check should probably be in the dependencies/databases file...
+                logger.warning(
+                    f"DatabaseWarning: {self.nr_fpath} needs to be formatted for diamond!"
+                )
+        else:
```

Comment:
This `if/else` is nesting variables that are not assigned under both `if` and `else` and therefore behavior should not change if they are not nested. This much nesting is also difficult to understand further indentation/behavior. This should be refactored for clarity